packages/internal-test-env/index.ts

```diff
@@ -0,0 +1,351 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+import { config } from "dotenv-flow";
+import { join } from "path";
+import { Session } from "@inrupt/solid-client-authn-node";
+import {
+  createContainerInContainer,
+  createSolidDataset,
+  deleteSolidDataset,
+  getPodUrlAll,
+  getSourceIri,
+  saveSolidDatasetInContainer,
+} from "@inrupt/solid-client";
+
+const availableEnvironment = [
+  "ESS Dev-Next" as const,
+  "ESS PodSpaces" as const,
+  "NSS" as const,
+  //  "ESS PodSpaces Next" as const,
+];
+
+export type AvailableEnvironment = typeof availableEnvironment extends Array<
+  infer E
+>
+  ? E
+  : never;
+
+const availableProtocol = ["ESS Notifications Protocol" as const];
+
+export type AvailableProtocol = typeof availableProtocol extends Array<infer E>
+  ? E
+  : never;
+
+export interface TestingEnvironmentNode {
+  environment: AvailableEnvironment;
+  idp: string;
+  notificationGateway: string | undefined;
+  protocol: AvailableProtocol | undefined;
+  clientCredentials: {
+    requestor: {
+      id: string;
+      secret: string;
+    };
+    resourceOwner: {
```

Comment:
```suggestion
    requestor: {
```

The "requestor" is the third-party, not the owner of the resources, so is the potentially undefined one.